### PR TITLE
snapcraft.yaml update with base:core20 

### PR DESCRIPTION
**Pull request by agrrr3**
The snapcraft.yaml was way outdated. This is a refresh, working with e.g. snapcraft 4.5

It is based on ubuntu 20.04 core and will hopefully be a good base for snapping the godot prototype.

Note that snap/snapcraft.yaml is already expanded, the unexpanded version is in 
snap/local/20210308_core20_unexpanded__snapcraft.yaml

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -207,7 +207,7 @@ jobs:
         - mkdir build
         - cd build
       script:
-        - cmake -GXcode -DBUILD_TESTING=OFF -DBoost_NO_BOOST_CMAKE=ON -DGLEW_USE_STATIC_LIBS=ON ..
```
Comment (geoffthemedio) [inline .travis.yml]: seems unrelated to topic?

File: test/parse/CMakeLists.txt [hunk 2]
```diff
@@ -19,6 +19,7 @@ target_include_directories(fo_unittest_parse
 
 target_link_libraries(fo_unittest_parse
     freeorioncommon
+    freeorionparse
```
Comment (geoffthemedio) [inline test/parse/CMakeLists.txt]: seems possibly unrelated to topic

File: .travis.yml (hunk 1 above)
Comment (agrrr3) [reply .travis.yml]: oops, seems i didnt edit the history correctly, thought b72fa4aa5 came from master. maybe from o01egs test branch. rebased. good to go.

**Comment by geoffthemedio** [comment]
I'll assume you know what you're doing, because I don't.
